XenModel/Actions/Pool/DestroyPoolAction.cs

```diff
@@ -33,25 +33,28 @@
 
 namespace XenAdmin.Actions
 {
-    public class DestroyPoolAction: PureAsyncAction
+    public class DestroyPoolAction: AsyncAction
     {
-        public DestroyPoolAction(XenAPI.Pool pool)
+        public DestroyPoolAction(Pool pool)
             : base(pool.Connection, string.Format(Messages.DESTROYING_POOL, pool.Name()))
         {
             System.Diagnostics.Trace.Assert(pool != null);
             Pool = pool;

```

Comment:
Ah, I didn't spot it earlier, but neither the assertion nor the null coalescing operator is useful because, if pool is null, it will crash before getting there: the line above calls the base constructor with pool.Connection and pool.Name(). I'll remove it completely.